### PR TITLE
feat: add `foldRightL` to List

### PR DESCRIPTION
**Pull request by nicdard**
Add implementation of foldRightL to List namespace together with the testcases. This first PR is used to check if the style and content is ok then I will implement the method in all the namespaces requested in the issue with another PR.

As discussed in #2877 I removed the polymorphic effect.

### CODE REVIEW

File: main/src/library/List.flix [hunk 1]
```diff
@@ -680,6 +680,18 @@ namespace List {
         case x :: rs => foldRightHelper(f, s, rs, s1 -> k(f(x, s1)))
     }
 
+    ///
+    /// Lazily applies `f` to a start value `s` and all  elements in `xs` going from right to left.
```
Comment (magnus-madsen) [inline main/src/library/List.flix:684]: extra space between all and elements.

File: main/src/library/List.flix [hunk 2]
```diff
@@ -680,6 +680,18 @@ namespace List {
...
+    ///
+    /// That is, the result is of the form: `f(x1, ...lazy f(xn-1, lazy f(xn, s))...)`.
+    ///
```
Comment (magnus-madsen) [inline main/src/library/List.flix:687]: /// The function `f` must be pure.

/// The foldRightL function exists to allow early termination of a fold.

File: main/test/ca/uwaterloo/flix/library/TestList.flix [hunk 3]
```diff
@@ -1023,6 +1023,21 @@ def foldRight03(): Bool = List.foldRight((e, i) -> (i - e)*(e rem 2 + 1), 100, 1
 @test
 def foldRight04(): Bool = List.foldRight((e, i) -> (i - e)*(e rem 2 + 1), 100, 1 :: 2 :: 3 :: Nil) == 382
 
+/////////////////////////////////////////////////////////////////////////////
+// foldRightL                                                              //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def foldRightL01(): Bool = List.foldRightL((e, i) -> (force i - e)*(e rem 2 + 1), 100, Nil) == 100
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/library/TestList.flix:1030]: space between *

**Review by magnus-madsen — COMMENTED** [review]
Looks good to me! Only have minor comments.

File: main/test/ca/uwaterloo/flix/library/TestList.flix (hunk 3 above)
Comment (nicdard) [reply main/test/ca/uwaterloo/flix/library/TestList.flix:1030]: I see, I was copying the style from other tests above, should I change them as well?

File: main/src/library/List.flix (hunk 1 above)
Comment (nicdard) [reply main/src/library/List.flix:684]: oh sorry for that

File: main/src/library/List.flix (hunk 2 above)
Comment (nicdard) [reply main/src/library/List.flix:687]: I assume I have to add those since you commented about the extra space, let me know if I need to remove anything

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**